### ISSUE TITLE
[Accessibility]: All date components should be using a number input

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/vets-design-system-documentation#6115]
### Point of contact

Robin Garrison

### How severe do you think this issue is?

High

### Severity rationale

See https://github.com/department-of-veterans-affairs/va.gov-team/issues/138628

> The day field allows voice control and screen reader users to input letters instead of restricting input to only numbers. This is a data invalidation error. VC users may find the allowance of both letters and numbers for this field to be confusing.

Though not explicitly stated, this problem is associated with both the `va-date` and `va-memorable-date` web components

### VA 508 Office

N/A

### URL where issue occurs

https://design.va.gov/storybook/?path=/docs/components-va-date--docs

### Component or pattern (if applicable)

- [va-date](https://design.va.gov/storybook/?path=/docs/components-va-date--docs)
- [va-memorable-date](https://design.va.gov/storybook/?path=/docs/uswds-va-memorable-date--docs)

### Environment details

- OS: All
- Browser: All
- Screen reader: All
- Device: All

### Steps to reproduce

1. Go to storybook
2. Enter letters in the month, day or year input
3. Or, try using Voice Control to enter content in date field

### Observed behavior

Non-numbers can be entered into the date fields

### Expected behavior / Acceptance criteria

- Only numbers should be allowed to be entered
- Only a number keyboard on mobile devices should be shown (already in place).

### Supporting evidence (screenshots, recordings, code snippets)

See https://github.com/department-of-veterans-affairs/va.gov-team/issues/138628

### Related issues or systemic notes

https://github.com/department-of-veterans-affairs/va.gov-team/issues/138628

### Type of issue

- [ ] Axe-core
- [x] Cognitive | user flows & navigating
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Content style guide
- [ ] Focus
- [ ] Headings
- [ ] Keyboard
- [ ] Link and button labels
- [ ] Markup and metadata
- [ ] Non-text content and media
- [x] Overall user experience
- [x] Screenreader | assistive tech & device support
- [ ] Semantic HTML
- [ ] URL standards
- [ ] Zoom or Magnification
- [ ] Something else
